### PR TITLE
Adding extra flu vaccine columns

### PR DESCRIPTION
**Pull request by rw251**
The PI for this study wants to compare COVID vaccine uptake with the typical flu vaccine uptake. Therefore I have added two columns. One for whether a person is "clinically" eligible for a vaccine, and one for when they received the flu vaccine in the 2019/2020 flu season.

### CODE REVIEW

File: projects/025 - Watkinson/template-sql/objectives-1-5.template.sql [hunk 1]
```diff
@@ -114,6 +118,8 @@ SELECT
 	CASE WHEN HighVulnerabilityCodeDate IS NOT NULL THEN 'Y' ELSE 'N' END AS HasHighClinicalVulnerabilityIndicator,
 	HighVulnerabilityCodeDate AS DateOfHighClinicalVulnerabilityIndicator,
 	CASE WHEN mv.FK_Patient_Link_ID IS NOT NULL THEN 'Y' ELSE 'N' END AS HasModerateClinicalVulnerability,
+	CASE WHEN flu.FK_Patient_Link_ID IS NOT NULL THEN 'Y' ELSE 'N' END AS IsClinicallyEligibleForFluVaccine,
+	FluVaccineDate AS DateOfFluVaccineIn20192020Season,
```
Comment (IliEle) [inline projects/025 - Watkinson/template-sql/objectives-1-5.template.sql:122]: worth adding a prefix on FluVaccineDate like below to be explicit on where it's coming from
```suggestion
	fluvac.FluVaccineDate AS DateOfFluVaccineIn20192020Season,
```

File: projects/025 - Watkinson/template-sql/objectives-1-5.template.sql [hunk 2]
```diff
@@ -131,5 +137,6 @@ LEFT OUTER JOIN #ModerateVulnerabilityPatients mv ON mv.FK_Patient_Link_ID = p.F
 LEFT OUTER JOIN #FirstCOVIDAdmission ca ON ca.FK_Patient_Link_ID = p.FK_Patient_Link_ID
 LEFT OUTER JOIN #COVIDDeath cd ON cd.FK_Patient_Link_ID = p.FK_Patient_Link_ID
 LEFT OUTER JOIN #COVIDVaccines v ON v.FK_Patient_Link_ID = p.FK_Patient_Link_ID
-LEFT OUTER JOIN #VaccineDeclinedPatients vd ON vd.FK_Patient_Link_ID = p.FK_Patient_Link_ID;
-
+LEFT OUTER JOIN #VaccineDeclinedPatients vd ON vd.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #FluVaccPatients flu ON flu.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientHadFluVaccine fluvac ON fluvac.FK_Patient_Link_ID = p.FK_Patient_Link_ID
```
Comment (IliEle) [inline projects/025 - Watkinson/template-sql/objectives-1-5.template.sql:142]: grain remains the same 👍

File: projects/025 - Watkinson/template-sql/objectives-1-5.template.sql [hunk 3]
```diff
@@ -65,6 +67,8 @@ SELECT * FROM #Temp;
 
 --> EXECUTE query-get-covid-vaccines.sql
 
+--> EXECUTE query-received-flu-vaccine.sql date-from:2019-07-01 date-to:2020-06-30
+
 --> EXECUTE query-get-flu-vaccine-eligible.sql
```
Comment (IliEle) [inline projects/025 - Watkinson/template-sql/objectives-1-5.template.sql:72]: has this been reviewed already? do you want me to have a look at this as well?

File: shared/Reusable queries for data extraction/query-received-flu-vaccine.sql [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+-- First get all patients from the GP_Events table who have a flu vaccination (procedure) code
+IF OBJECT_ID('tempdb..#PatientsWithFluVacConcept') IS NOT NULL DROP TABLE #PatientsWithFluVacConcept;
+SELECT FK_Patient_Link_ID, CAST(EventDate AS DATE) AS FluVaccineDate
+INTO #PatientsWithFluVacConcept
+FROM RLS.[vw_GP_Events]
+WHERE (
+	FK_Reference_Coding_ID IN (SELECT FK_Reference_Coding_ID FROM #VersionedCodeSets WHERE Concept = 'flu-vaccination' AND [Version] = 1) OR
+	FK_Reference_SnomedCT_ID IN (SELECT FK_Reference_SnomedCT_ID FROM #VersionedSnomedSets WHERE Concept = 'flu-vaccination' AND [Version] = 1)
+)
+AND EventDate > '{param:date-from}'
```
Comment (IliEle) [inline shared/Reusable queries for data extraction/query-received-flu-vaccine.sql:32]: Shouldn't this include the date-from as mentioned in documentation - line 8? 
```suggestion
AND EventDate >= '{param:date-from}'
```

File: shared/Reusable queries for data extraction/query-received-flu-vaccine.sql [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+--> CODESETS flu-vaccine
+-- Then get all patients from the GP_Medications table who have a flu vaccine (medication) code
+INSERT INTO #PatientsWithFluVacConcept
+SELECT FK_Patient_Link_ID, CAST(MedicationDate AS DATE) FROM RLS.vw_GP_Medications
+WHERE (
+	FK_Reference_Coding_ID IN (SELECT FK_Reference_Coding_ID FROM #VersionedCodeSets WHERE Concept = 'flu-vaccine' AND [Version] = 1) OR
+	FK_Reference_SnomedCT_ID IN (SELECT FK_Reference_SnomedCT_ID FROM #VersionedSnomedSets WHERE Concept = 'flu-vaccine' AND [Version] = 1)
+)
+and MedicationDate > '{param:date-from}'
```
Comment (IliEle) [inline shared/Reusable queries for data extraction/query-received-flu-vaccine.sql:43]: Same as above.

**Review by IliEle — CHANGES_REQUESTED** [review]
Overall looking good. Some minor suggestions and comments below. 
Liked the use of code sets within the template file, didn't know we could do that.

File: projects/025 - Watkinson/template-sql/objectives-1-5.template.sql (hunk 2 above)
Comment (rw251) [reply projects/025 - Watkinson/template-sql/objectives-1-5.template.sql]: Thanks - change made.

File: shared/Reusable queries for data extraction/query-received-flu-vaccine.sql (hunk 5 above)
Comment (rw251) [reply shared/Reusable queries for data extraction/query-received-flu-vaccine.sql]: Change made. Thanks.

File: projects/025 - Watkinson/template-sql/objectives-1-5.template.sql (hunk 3 above)
Comment (rw251) [reply projects/025 - Watkinson/template-sql/objectives-1-5.template.sql:72]: Yes please - though you spotted that the > should be >= in the mentioned file so I assume you did actually look at this in the end. Is that right?

Comment (rw251) [reply projects/025 - Watkinson/template-sql/objectives-1-5.template.sql:72]: No that file has already been reviewed by George.

**Pull request merged**